test_cases/test_easygui.py

```diff
@@ -0,0 +1,124 @@
+    assert os.path.exists(os.path.join(FOLDER_OF_THIS_FILE, 'result.png'))
+
+def test_msgbox():
+    # Test hitting space to click OK.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox() == 'OK'
+
+    # Test hitting Esc to close.
+    t = KeyPresses([Key.esc])
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox() is None
+
+    # Test with custom message.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox('Message') == 'OK'
+
+    # Test with custom message and title.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox('Message', 'Title') == 'OK'
+
+    # Test with custom OK button text.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox(ok_button='Button') == 'Button'
+
+    # Test with custom message, title, and OK button text.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox('Message', 'Title', ok_button='Button') == 'Button'
+
+    # Test with jpg image.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox(image=os.path.join(FOLDER_OF_THIS_FILE, 'pi.jpg')) == 'OK'
+
+    # Test with png image.
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.msgbox(image=os.path.join(FOLDER_OF_THIS_FILE, 'result.png')) == 'OK'
+
+def test_buttonbox():
+    # Test hitting space to click OK with different default buttons:
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.buttonbox('Message', 'Title', choices=('Button[1]', 'Button[2]', 'Button[3]'), default_choice='Button[1]') == 'Button[1]'
+
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.buttonbox('Message', 'Title', choices=('Button[1]', 'Button[2]', 'Button[3]'), default_choice='Button[2]') == 'Button[2]'
+
+    t = KeyPresses(' ')
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.buttonbox('Message', 'Title', choices=('Button[1]', 'Button[2]', 'Button[3]'), default_choice='Button[3]') == 'Button[3]'
+
+    # Test hitting Esc to close.
+    # TODO: If button boxes aren't given a default choice, then their window won't be in focus and this test hangs.
+    #t = KeyPresses([Key.esc])
+    #t.start()
+    #print('Line', inspect.currentframe().f_lineno)
+    #assert easygui.buttonbox() is None
+
+    # Test hitting Esc to close.
+    t = KeyPresses([Key.esc])
+    t.start()
+    print('Line', inspect.currentframe().f_lineno)
+    assert easygui.buttonbox(default_choice='Button[1]') is None
+
```

Comment:
Same idea here... also marked the TODO item test as an `xfail` instead of commenting it out.

```suggestion
@pytest.mark.parametrize('position', (0, 1, 2))
def test_default_choice_is_selected(position):
    # Test hitting space to click OK with different default buttons:
    t = KeyPresses(' ')
    t.start()
    choices = ('Button[1]', 'Button[2]', 'Button[3]')
    expected = choices[position]
    print('Line', inspect.currentframe().f_lineno)
    assert easygui.buttonbox('Message', 'Title', choices=choices, default_choice=expected) == expected

# TODO: If button boxes aren't given a default choice, then their window won't be in focus and this test hangs.
@pytest.mark.xfail(strict=True)  
def test_hitting_escape_presses_close()
    t = KeyPresses([Key.esc])
    t.start()
    box = easygui.buttonbox(run=False)
    def killbox():
        # if pynput failed, makes sure the dialog is closed to try to prevent deadlock
        try:
            box.ui.boxRoot.destroy()
        except:
            pass
    box.ui.boxRoot.after(5000, killbox)
    assert easygui.buttonbox() is None

def test_hitting_esc_closes_with_default_choice():
    # Test hitting Esc to close.
    t = KeyPresses([Key.esc])
    t.start()
    assert easygui.buttonbox(default_choice='Button[1]') is None

```

Reply:
Sounds good to me.